src/GafferImage/ImageWriter.cpp

```diff
@@ -869,6 +935,12 @@ IECore::MurmurHash ImageWriter::hash( const Context *context ) const
 void ImageWriter::execute() const
 {
 	// Create an OIIO::ImageOutput
+	Context * c = const_cast<Context *>( Context::current() );
+	c->set( "gaffer:colorspace", defaultColorSpace()( currentFileFormat() ) );
+
+
+	Expression *e = const_cast<Expression *>( expression() );
```

Comment:
Setting the expression is classed as a graph edit, and it's forbidden to make those from within `execute()` (because threaded execution could mean that another call to `execute()` is already in flight). From our previous discussions I know why we can't put this in the constructor where we'd like it though, so I'm not 100% sure what to do. One thought would be to make a very simple ComputeNode to return the value from the context, so we don't need the Expression node at all. Although that's a bit more code, it would be nice to remove the dependency on Expression (and Python) from ImageWriter. Don't worry about this immediately, but I think I might take a look at it before we merge, or propose that we do it soon after.